### DISCUSSION TITLE
feature request: Add support for csvq

### DISCUSSION THREAD

**Discussion by jftuga** [grafana/grafana-csv-datasource#177] [Ideas]
Please consider adding support for [csvq](https://github.com/mithrandie/csvq).  This will allow for using an SQL-like query language for CSV data.

One benefit from would include the ability to add a delta column to the CSV table using SQL `LAG()` or `LEAD()` functions.

Example:

Original CSV, named `test.csv`:

```
Date, Processed
2021-04-15, 300
2021-04-14, 275
```

Augmented output with `csvq` could then be rendered in `Grafana` like this:
```
csvq 'select Date, Processed, LAG(Processed) over (order by Date) as Previous, (Processed-Previous) as Delta from test order by Date desc'
```

|    Date    | Processed | Previous | Delta |
|------------|-----------|----------|-------|
| 2021-04-15 |  300      |  275     |    25 |
| 2021-04-14 |  275      |   NULL   | NULL  |

**Comment by marcusolsson** [comment]
Thank you for the feature request! I think this could be really useful in giving users more control over the data returned.

**Comment by HenrikRolandHansen** [comment]
This would be a nice feature instead of using filtering as suggested in #116. adding support for using variables with csvq would be handy as well.
